File: app/src/main/java/com/kamelong/OuDia/StationTrack.java
package com.kamelong.OuDia;

import java.io.PrintWriter;
/*
 * Copyright (c) 2019 KameLong
 * contact:kamelong.com
 *
 * This source code is released under GNU GPL ver3.
 */

/**
 * 番線１つを表します。
 * 全てのStationTrackはStationによって管理されます
 */
public class StationTrack implements Cloneable{
    /**
     番線名
     */
    public String trackName="";
    /**
     番線略称(共通or下り)
     */
    public String trackShortName="";
    /**
     番線略称(上り)
     この要素は空白も可能です。
     空白の場合、m_strTrackRyakusyouが上下両方に対応します。
     */
    public String trackShortNameUp="";

    public StationTrack(String name,String shortName){
        trackName=name;
        trackShortName=shortName;
        trackShortNameUp="";
    }
    public StationTrack(){

    }
    /**
     * oudiaファイルを1行読み込む
     */
    void setValue(String title,String value){
        switch (title){
            case"TrackName":
                trackName=value;
                break;
            case"TrackRyakusyou":
                trackShortName=value;
                break;
            case"TrackNoboriRyakusyou":
                trackShortNameUp=value;
                break;
        }
    }

    /**
     * oudia2nd 形式でファイルを保存する
     */
    void saveToFile(PrintWriter out){
        out.println("EkiTrack2.");
        out.println("TrackName="+trackName);
        out.println("TrackRyakusyou="+trackShortName);
        if(trackShortNameUp.length()!=0) {
            out.println("TrackNoboriRyakusyou=" + trackShortNameUp);
        }
        out.println(".");
    }
    @Override
    public StationTrack clone() {
        try {
            return (StationTrack) super.clone();
        } catch (CloneNotSupportedException e) {
            e.printStackTrace();
        }
        return new StationTrack();
    }
    /*
    =================================
    ここまでOuDiaライブラリ共通の処理
    =================================
     */


}

File: app/src/main/java/com/kamelong/OuDia/StationTrack.kt
package com.kamelong.OuDia

import java.io.PrintWriter

/*
 * Copyright (c) 2019 KameLong
 * contact:kamelong.com
 *
 * This source code is released under GNU GPL ver3.
 */
/**
 * 番線１つを表します。
 * 全てのStationTrackはStationによって管理されます
 */
class StationTrack : Cloneable {
    /**
     * 番線名
     */
    var trackName = ""

    /**
     * 番線略称(共通or下り)
     */
    var trackShortName = ""

    /**
     * 番線略称(上り)
     * この要素は空白も可能です。
     * 空白の場合、m_strTrackRyakusyouが上下両方に対応します。
     */
    var trackShortNameUp = ""

    constructor(name: String, shortName: String) {
        trackName = name
        trackShortName = shortName
        trackShortNameUp = ""
    }

    constructor() {}

    /**
     * oudiaファイルを1行読み込む
     */
    fun setValue(title: String?, value: String) {
        when (title) {
            "TrackName" -> trackName = value
            "TrackRyakusyou" -> trackShortName = value
            "TrackNoboriRyakusyou" -> trackShortNameUp = value
        }
    }

    /**
     * oudia2nd 形式でファイルを保存する
     */
    fun saveToFile(out: PrintWriter) {
        out.println("EkiTrack2.")
        out.println("TrackName=$trackName")
        out.println("TrackRyakusyou=$trackShortName")
        if (trackShortNameUp.length != 0) {
            out.println("TrackNoboriRyakusyou=$trackShortNameUp")
        }
        out.println(".")
    }

    public override fun clone(): StationTrack {
        try {
            return super.clone() as StationTrack
        } catch (e: CloneNotSupportedException) {
            e.printStackTrace()
        }
        return StationTrack()
    } /*
    =================================
    ここまでOuDiaライブラリ共通の処理
    =================================
     */
}